CONTRIBUTING.md

```diff
@@ -0,0 +1,856 @@
+
+# Contributing
+
+We welcome any and all contributions. This is a community-driven project. Although we don't have a lot, we do have a few guidelines for contributions.
+
+1. Please adhere to the style guide. It's pretty lengthy, but most everything is checked by ESLint, and if anything fails to check, the build will fail, and you will know about it. Most of it is common practice and common sense. ESLint is also set up to check for other common errors, such as undeclared variables and invalid `typeof` values.
+2. Please make sure there are no regressions with your patch. Please don't disable existing tests, and please don't send a PR with new, disabled tests.
+	- There are a few known failing tests currently (PRs welcome): [*](https://github.com/lhorie/mithril.js/blob/next/test/mithril.deferred.js#L121-L125) [*](https://github.com/lhorie/mithril.js/blob/next/test/mithril.render.js#L1321-L1343) [*](https://github.com/lhorie/mithril.js/blob/next/test/mithril.route.js#L106-L131) [*](https://github.com/lhorie/mithril.js/blob/next/test/mithril.route.js#L134-L162) [*](https://github.com/lhorie/mithril.js/blob/next/test/mithril.route.js#L165-L191) [*](https://github.com/lhorie/mithril.js/blob/next/test/mithril.route.js#L194-L222)
+3. For any new features introduced, be sure to write new unit tests for it. Maximum coverage is what we want.
+4. Try to not leave any extra `TODO`s, `FIXME`s, etc. in your code. ESLint will nag at you until you fix whatever problem it is.
+	- Note that it's only a warning, not an error. It won't fail the CI tests, and there's a few outstanding ones inside Mithril right now.
+
+It is assumed that all contributions you make you have the appropriate rights to and that it may be made available under the MIT License, the license used for this project.
+
+# Style Guide
+
+The style is checked with ESLint. This style guide is here for one reason only: consistency. This should work for most code here, but this shouldn't be considered absolute &endash; consistency with the surrounding code is higher priority than following this guide. Also, if some sort of hack is impossible without violating this guide (e.g. if Mithril ever needs [Bluebird's infamous hack](https://stackoverflow.com/q/24987896)), just make sure the code is consistent with what's around it.
+
+### Line length
+
+Keep lines down to a maximum of 80 characters. Minifiers are very good at compressing whitespace, so don't be afraid to use it.
+
+The only exception to this rule is with long regexes. Append `// eslint-disable-line max-len` to the end of those lines.
+
+### Function length
+
+Keep function length to no more than 20 statements, including those in nested blocks.
+
+The only exceptions are for revealing module wrappers and Mocha `describe` and `context` blocks.
+
+This isn't checked for the tests, but still, keep it reasonable.
+
+### Line endings
+
+Use Windows-style line endings (i.e. CRLF).
```

Comment:
That is a question for @lhorie. Git is set to convert everything automatically through its .gitattributes file, and he also added a .editorconfig to do this. It also makes it easier to locally develop, as many diff algorithms don't take into account the line endings.

The main argument for it is better compatibility with Windows programs, although that is about 50% bullshit nowadays. The main argument against is for easier parsing, which unless you're doing much in text processing, it's also pretty invalid.